File: src/main/java/net/matees/settings/StringSetting.java
package net.matees.settings;

import java.util.List;

public abstract class StringSetting extends Setting<String> {
    @Override
    public void setIntValue(int value) {  }

    @Override
    public void setBooleanValue(boolean value) { }

    @Override
    public void setStringValue(String value) { setSetting(value); }

    @Override
    public void setListValue(List value) { }

}

File: src/main/java/net/matees/settings/StringSetting.kt
package net.matees.settings

import org.bukkit.entity.Player

abstract class StringSetting : Setting<String?>() {
    override fun setIntValue(value: Int) {
    }

    override fun setBooleanValue(value: Boolean) {
    }

    override fun setStringValue(value: String?) {
        setting = value
    }

    override fun setPlayersValue(value: List<Player?>?) {
    }
}
